controller_manager/src/controller_manager.cpp

```diff
@@ -1092,9 +1096,14 @@ controller_interface::return_type ControllerManager::unload_controller(
     return controller_interface::return_type::ERROR;
   }
 
+  // find and clean controller if it is inactive
+  if (cleanup_controller(controller_name) != controller_interface::return_type::OK)
+  {
+    return controller_interface::return_type::ERROR;
+  }
```

Comment:
```suggestion
  if (is_controller_inactive(*controller.c) && (cleanup_controller(controller_name) != controller_interface::return_type::OK))
  {
    return controller_interface::return_type::ERROR;
  }
```

Should we check if it is inactive before doing the cleanup? I mean it would also be in unconfigured state right?